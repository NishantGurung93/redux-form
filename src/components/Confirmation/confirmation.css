.confirm-checkmark {
  font-size: 4rem;
  color: green;
  text-align: center;
}

.confirm-message {
  width: 100%;
  color: green;
  text-align: center;
}
