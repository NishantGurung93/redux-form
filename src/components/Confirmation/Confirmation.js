import { useSelector } from "react-redux";
import './confirmation.css';

const Confirmation = () => {
  const reduxState = useSelector(state => state);
  console.log(JSON.stringify(reduxState));

  return (
    <>
      <div className="confirm-checkmark">&#10003;</div>
      <div className="confirm-message">Please verify your email address, you should have received an email from us already!</div>
    </>
  )
};

export default Confirmation;
