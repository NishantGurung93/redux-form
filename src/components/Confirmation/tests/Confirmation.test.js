import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import Confirmation from "../Confirmation";

const mockStore = {
  value: {},
  useDispatch: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(),
  subscribe: jest.fn(),
}

console.log = jest.fn();

const renderComponent = () => render(
  <Provider store={mockStore}>
    <Confirmation />
  </Provider>
);

describe('Confirmation component', () => {
  it('should render the component', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
