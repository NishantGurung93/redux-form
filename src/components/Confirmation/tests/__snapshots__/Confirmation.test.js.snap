// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation component should render the component 1`] = `
<div>
  <div
    class="confirm-checkmark"
  >
    âœ“
  </div>
  <div
    class="confirm-message"
  >
    Please verify your email address, you should have received an email from us already!
  </div>
</div>
`;
